apps/test/netsim/NetSimVisualization.js

```diff
@@ -84,15 +89,74 @@ describe("NetSimVisualization", function () {
   var getVizNodeByEntityID_ = function (_type, id) {
     return this.filter(function(element){
       return element instanceof NetSimVizNode &&
-          element.getCorrespondingEntityID &&
-          element.getCorrespondingEntityID() === id;
+          element.getCorrespondingEntityId &&
+          element.getCorrespondingEntityId() === id;
     })[0];
   };
 
+  function createNode(shard, name) {
+    var newNode;
+    NetSimLocalClientNode.create(shard, name, function (e, n) {
+      newNode = n;
+    });
+    assert(newNode !== undefined, "Failed to create a client.");
+    return newNode;
+  }
+
+  it("does not mismap elements across shard reset", function () {
+    // Fake NetSim
+    var netsim = new NetSim();
+    netSimVis = new NetSimVisualization(makeRootDiv(), netsim.runLoop_);
+
+    // Connect to shard
+    testShard = fakeShard();
+    var myNode = createNode(testShard, 'myNode');
+    netSimVis.setShard(testShard);
+    netSimVis.setLocalNode(myNode);
+
+    // Make a couple more nodes, and make sure the visualization picks them up.
+    var node2 = createNode(testShard, 'node2');
+    var node3 = createNode(testShard, 'node3');
+
+    // Check out the list of viz elements
+    assert.equal(3, netSimVis.elements_.length);
+
+    // Introduce a new shard, with new data
```

Comment:
It looks like you are creating `newNode`, adding it to the visualization, then creating `newNode2` and... not adding it to the visualization? It would be helpful to explain why you are treating the two nodes differently here.